### PR TITLE
Remove clearing the block selection on sidebar tab switch

### LINKED ISSUE #18239: Preserve selection when switching block editor tabs
**Describe the bug**
This is probably an intended feature, but when switching tabs in the editor sidebar between document and block, the current block selection is lost.

The problem is easily recoverable on desktop, but it presents an almost empty screen on mobile, looking like an error.

**To reproduce**
1. Selet a block
2. Show the sidebar
3. Click on Document
4. Click on Block
5. No block is selected so this panel is now empty

**Expected behavior**
I expect the selection to be preserved because there is nothing I did to it by fiddling with the sidebar's tabs.

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Closes #18239

## How has this been tested?
Tested locally.

## Screenshots
no-selection-clearing

## Types of changes
It is a simple update of the `settings-header` that removes the call to `clearSelectedBlock`.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Review by jorgefilipecosta — COMMENTED** [review]
Hi, this works as expected from the code point of view. But I would wait on feedback from @jasmussen as the original author of https://github.com/WordPress/gutenberg/issues/6887 before merging.

**Comment by jasmussen** [comment]
Thanks. I've shared with Andrei that I think this is one of those things that, if it's a trivial amount of code, we can just go ahead and try it. I'm still not convinced it's actually preferable behavior, but if we merge and dislike, we can always revert. So no blockers from me!

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

**Pull request merged**
